src/theory/arith/nl/iand_utils.cpp

```diff
@@ -146,7 +154,35 @@ Node IAndTable::createBitwiseNode(Node x,
   return sumNode;
 }
 
-void IAndTable::computeAndTable(uint64_t granularity)
+Node IAndUtils::createBitwiseIAndNode(Node x,
+                                      Node y,
+                                      uint64_t high,
+                                      uint64_t low)
+{
+  uint64_t granularity = high - low + 1;
+  Assert(granularity <= 8);
+  // compute the table for the current granularity if needed
+  if (d_bvandTable.find(granularity) == d_bvandTable.end())
+  {
+    computeAndTable(granularity);
+  }
+  const std::map<std::pair<int64_t, int64_t>, uint64_t>& table =
+      d_bvandTable[granularity];
+  return createITEFromTable(
```

Comment:
I am a bit confused. in the bit-wise mode, isn't the idea to have a big conjunction rather than a big ite?

Reply:
I'm not sure tbh, in the original branch that we based this on, it was an ITE. What were you envisioning for a big conjunction?